rust/Cargo.toml

```diff
@@ -105,6 +106,11 @@ s3-rustls = [
 ]
 glue = ["s3", "rusoto_glue"]
 python = ["arrow/pyarrow"]
+
+# AWS_PROFILE is experimental in object_store, and comes with a hefty load of dependencies.
+# must be used in conjunction with s3 or s3-rustls feature
+aws-profile = ["object_store/aws_profile"]
```

Comment:
Agreed, this would have been better in a separate PR - here I was working in WSL and struggling with some memory issues, which is why this change made my live a bit easier there and then 😆.

Reply:
Just realized, that the current setup is likely slightly broken anyhow. IF we include the feature in the object-store crate by default, this will enable aws / s3 support in the object store crate as well, however we are not enabling either s3 or s3-rustls by default on our end, in which case delta.-rs will not be able to make use of these features.

Reply:
Following the previous comment, we could set this `aws-profile` feature with S3 features since it's only related to the AWS S3 backend storage.